COMETwebapp/Services/RowViewModelFactoryService/RowViewModelFactory.cs

```
@@ -0,0 +1,87 @@
+                MeasurementUnit castedThing => new MeasurementUnitRowViewModel(castedThing),
+                ParameterTypeComponent castedThing => new ParameterTypeComponentRowViewModel(castedThing),
+                ParameterType castedThing => new ParameterTypeRowViewModel(castedThing),
+                QuantityKindFactor castedThing => new QuantityKindFactorRowViewModel(castedThing),
+                ScaleValueDefinition castedThing => new ScaleValueDefinitionRowViewModel(castedThing),
+                UnitFactor castedThing => new UnitFactorRowViewModel(castedThing),
+
+                // Site directory rows
+                DomainOfExpertise castedThing => new DomainOfExpertiseRowViewModel(castedThing),
+                EngineeringModelSetup castedThing => new EngineeringModelRowViewModel(castedThing),
+                Iteration castedThing => new IterationRowViewModel(castedThing),
+                OrganizationalParticipant castedThing => new OrganizationalParticipantRowViewModel(castedThing),
+                Organization castedThing => new OrganizationRowViewModel(castedThing),
+                ParticipantRole castedThing => new ParticipantRoleRowViewModel(castedThing),
+                Participant castedThing => new ParticipantRowViewModel(castedThing),
+                PersonRole castedThing => new PersonRoleRowViewModel(castedThing),
+                Person castedThing => new PersonRowViewModel(castedThing),
+
+                // Engineering model rows
+                CommonFileStore castedThing => new CommonFileStoreRowViewModel(castedThing),
+                FileRevision castedThing => new FileRevisionRowViewModel(castedThing),
+                FileType castedThing => new FileTypeRowViewModel(castedThing),
+                Option castedThing => new OptionRowViewModel(castedThing),
+                Publication castedThing => new PublicationRowViewModel(castedThing),
+                _ => null
+            };
```

I would raise a NonSupportedException instead of returning null